db/music.dbml

```diff
@@ -0,0 +1,299 @@
+  song_id int [not null, ref: > song.song_id]
+  order int [not null]
+  date_submitted timestamp [not null, default:`now()`]
+  active bool_int [not null]
+  is_dupe bool_int [not null]
+  title text [not null]
+  version text [null]
+  artist text [null]
+  collaborators text [null]
+  composer text [null]
+  link text [not null]
+  genre text [null]
+  emoji text [null]
+  comments text [null]
+
+  indexes {
+    submission_id
+    participant_id
+    bracket_id
+    song_id
+  }
+
+  Note: 'composite unique index on submission_id, date_function (https://stackoverflow.com/a/26304364/2506158)'
+}
+
+Table link_relation {
+  link_realtion_id uuid [pk, default: `gen_random_uuid()`]
+  relation_id int [not null]
+  link_id int [not null, ref: > link.link_id]
+  type_link text [not null, Note: 'submission song seed' ]
+
+  indexes {
+    relation_id
+  }
+}
+
+Table link {
+  link_id uuid [pk, default: `gen_random_uuid()`]
+  url text [not null]
+  region text[] [null, Note: 'US Canada Japan Australia Europe Asia "North America" "South America" Africa Worldwide' ]
+}
+
+Table song {
+  song_id uuid [pk, default: `gen_random_uuid()`]
+  title text [not null]
+  title_english text [null]
+  version text [null]
+  artist text [not null, Note: 'CALCULATED']
+  collaborators text [null, Note: 'CALCULATED']
+  composer text [null]
+  year_min int [null]
+  year_max int [null]
+
+  indexes {
+    (title, version, artist, collaborators) [unique]
+  }
+}
+
+Table song_relation {
+  song_relation_id int [pk,increment]
+  song_from_id int [not null]
+  song_to_id int [not null]
+  relation_type text [not null, Note: 'cover live' ]
+}
+
+Table song_artist {
+  song_artist_id uuid [pk, default: `gen_random_uuid()`]
+  song_id int [not null, ref: > song.song_id]
+  artist_id int [not null, ref: > artist.artist_id]
+  credit_type text [not null, Note: 'primary collaborator original remixer composer' ]
+
+  indexes {
+    (song_id, artist_id) [unique]
+  }
+}
+
+Table artist {
+  artist_id uuid [pk, default: `gen_random_uuid()`]
+  name text [not null]
+  year_min int [null]
+  year_max int [null]
+  primary_artist_id int [null, ref: > artist.artist_id]
+}
+
+Table artist_genre {
+  artist_genre_id uuid [pk, default: `gen_random_uuid()`]
+  artist_id int [not null, ref: > artist.artist_id]
+  genre_id int [not null, ref: > genre.genre_id]
+}
+
+Table participant {
+  participant_id uuid [pk, default: `gen_random_uuid()`]
+  username text [unique, not null]
+  nickname text [null, Note: 'CALCULATED']
+  date_joined timestamp [not null, default:`now()`]
+  time_zone text [null]
+  active bool_int [not null]
+}
+
+Table participant_nickname {
+  participant_nickname_id uuid [pk, default: `gen_random_uuid()`]
+  participant_id int [not null, ref: > participant.participant_id]
+  nickname text [not null]
+  date_start date [not null]
+  date_end date [null]
+
+  indexes {
+    participant_id
+  }
+  
+  Note: 'composite unique index on participant_id, date_function'
```

Comment:
I think we might want a composite index on `(participant_id, date_start)` depending on how we feel about my other comment re `date_end`, but I might not make it unique. I guess if we *do* somehow try to have multiple nicknames for a single participant with the exact same datetime on them, it would be problematic for knowing what the "current" nickname is, we really do want to have a strict ordering on these rows, one way or another.

Reply:
Sounds good.